src/lib/utils/ct_utils.h

```diff
@@ -96,30 +96,36 @@ inline bool poison_has_effect() {
  * Poison a single integral object
  */
 template <std::integral T>
-constexpr void poison(T& p) {
+constexpr T& poison(T& p) {
    poison(&p, 1);
+   return p;
 }
 
 template <std::integral T>
-constexpr void unpoison(T& p) {
+constexpr T& unpoison(T& p) {
    unpoison(&p, 1);
+   return p;
 }
 
 /**
  * Poison a contiguous buffer of trivial objects (e.g. integers and such)
  */
 template <ranges::spanable_range R>
-   requires std::is_trivially_copyable_v<std::ranges::range_value_t<R>>
-constexpr void poison(R&& r) {
+   requires std::is_trivially_copyable_v<std::ranges::range_value_t<R>> &&
+            (!requires(const R& x) { x._const_time_poison(); })
+constexpr R&& poison(R&& r) {
```

Comment:
Do we have class that adhere to `spanable_range` and have the `const_time_poison()` helpers? Or what do we need the new `requires` for?

Reply:
I still do not see a point in `poison` returning anything

Reply:
I had a problem with the Crystal Polynomials, which fulfill the range concept AND have the const_time_poison methods (and are seemingly trivially copyable?). After implementing the pass-through, I encountered compiler errors, claiming that `poison` was ambiguous. I decided that the `const_time_poison()` method should always take precedence in such cases. I wonder why this did not occur without the pass-through, though...

Reply:
> I still do not see a point in poison returning anything

@randombit I feel the discussion about the general utility of this PR encountered a bit of a race condition. 😏
To avoid misunderstandings, let's keep the "yay or nay" discussion to the main thread. I'm not sure you've seen my argument about [restricting this pass-through to rvalue parameters](https://github.com/randombit/botan/pull/4260#issuecomment-2250223444).